PR: https://github.com/odoo/odoo/pull/31243

From: 3cd7ed07a29c89ddf193796c20a812b9bf21e284
From: Hiral Bhavsar
Date: 2019-06-17 11:34:17

Structural Changes: 1
Total Changes: 74

[IMP] *: remove 'view_type' on window actions.

The old tree views don't really exist anymore, this odd pseudo-flag to
dispatch between "list" and "tree" tree views has no reason to remain.

Task 1937686

closes odoo/odoo#31243

Signed-off-by: Xavier Morel (xmo) <xmo@odoo.com>

================================= pseudo patch: =================================

--- a/odoo/addons/base/models/ir_actions.py
+++ b/odoo/addons/base/models/ir_actions.py
@@ -183,8 +183,6 @@ class IrActionsActWindow(models.Model):
     target = fields.Selection([('current', 'Current Window'), ('new', 'New Window'), ('inline', 'Inline Edit'), ('fullscreen', 'Full Screen'), ('main', 'Main action of Current Window')], default="current", string='Target Window')
     view_mode = fields.Char(required=True, default='tree,form',
                             help="Comma-separated list of allowed view modes, such as 'form', 'tree', 'calendar', etc. (Default: tree,form)")
-    view_type = fields.Selection([('tree', 'Tree'), ('form', 'Form')], default="form", string='View Type', required=True,
-                                 help="View type: Tree type to use for the tree view, set to 'tree' for a hierarchical tree view, or 'form' for a regular list view")
     usage = fields.Char(string='Action Usage',
                         help="Used to filter menu and home actions from the user form.")
     view_ids = fields.One2many('ir.actions.act_window.view', 'act_window_id', string='No of Views')

--- a/odoo/addons/base/models/ir_actions_report.py
+++ b/odoo/addons/base/models/ir_actions_report.py
@@ -765,7 +765,6 @@ class IrActionsReport(models.Model):
                 'name': _('Choose Your Document Layout'),
                 'type': 'ir.actions.act_window',
                 'context': {'default_report_name': self.report_name, 'discard_logo_check': True},
-                'view_type': 'form',
                 'view_mode': 'form',
                 'res_id': self.env.company.id,
                 'res_model': 'res.company',

--- a/odoo/addons/base/models/ir_fields.py
+++ b/odoo/addons/base/models/ir_fields.py
@@ -306,7 +306,7 @@ class IrFieldsConverter(models.AbstractModel):
         warnings = []
         error_msg = ''
         action = {'type': 'ir.actions.act_window', 'target': 'new',
-                  'view_mode': 'tree,form', 'view_type': 'form',
+                  'view_mode': 'tree,form',
                   'views': [(False, 'tree'), (False, 'form')],
                   'help': _(u"See all possible values")}
         if subfield is None:

--- a/odoo/addons/base/models/ir_module.py
+++ b/odoo/addons/base/models/ir_module.py
@@ -33,7 +33,6 @@ from odoo.http import request
 _logger = logging.getLogger(__name__)
 
 ACTION_DICT = {
-    'view_type': 'form',
     'view_mode': 'form',
     'res_model': 'base.module.upgrade',
     'target': 'new',

--- a/odoo/addons/base/models/res_users.py
+++ b/odoo/addons/base/models/res_users.py
@@ -726,7 +726,6 @@ class Users(models.Model):
         self.ensure_one()
         return {
             'name': _('Groups'),
-            'view_type': 'form',
             'view_mode': 'tree,form',
             'res_model': 'res.groups',
             'type': 'ir.actions.act_window',
@@ -739,7 +738,6 @@ class Users(models.Model):
         self.ensure_one()
         return {
             'name': _('Access Rights'),
-            'view_type': 'form',
             'view_mode': 'tree,form',
             'res_model': 'ir.model.access',
             'type': 'ir.actions.act_window',
@@ -752,7 +750,6 @@ class Users(models.Model):
         self.ensure_one()
         return {
             'name': _('Record Rules'),
-            'view_type': 'form',
             'view_mode': 'tree,form',
             'res_model': 'ir.rule',
             'type': 'ir.actions.act_window',

--- a/odoo/addons/base/views/ir_actions_views.xml
+++ b/odoo/addons/base/views/ir_actions_views.xml
@@ -40,7 +40,6 @@
             <field name="name">Actions</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">ir.actions.actions</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="action_view_tree"/>
             <field name="search_view_id" ref="action_view_search"/>
         </record>
@@ -129,7 +128,6 @@
             <field name="name">Reports</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">ir.actions.report</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="act_report_xml_view_tree"/>
             <field name="search_view_id" ref="act_report_xml_search_view"/>
         </record>
@@ -144,7 +142,6 @@
                 <tree string="Open Window">
                     <field name="name"/>
                     <field name="res_model"/>
-                    <field name="view_type"/>
                     <field name="view_id"/>
                     <field name="domain"/>
                     <field name="context"/>
@@ -218,10 +215,6 @@
             <field name="arch" type="xml">
                 <search string="Open a Window">
                     <field name="name" filter_domain="['|', ('name','ilike',self), ('res_model','ilike',self)]" string="Action"/>
-                    <field name="view_type"/>
-                    <group expand="0" string="Group By" colspan="4">
-                        <filter string="View Type" name="view_type" domain="[]" context="{'group_by':'view_type'}"/>
-                    </group>
                 </search>
             </field>
         </record>
@@ -229,7 +222,6 @@
             <field name="name">Window Actions</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">ir.actions.act_window</field>
-            <field name="view_type">form</field>
             <field name="search_view_id" ref="view_window_action_search"/>
         </record>
         <record id="ir_action_window_view1" model="ir.actions.act_window.view">
@@ -387,7 +379,6 @@ env['res.partner'].create({'name': partner_name})
             <field name="name">Server Actions</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">ir.actions.server</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
             <field name="view_id" ref="view_server_action_tree"/>
             <field name="search_view_id" ref="view_server_action_search"/>
@@ -451,7 +442,6 @@ env['res.partner'].create({'name': partner_name})
             <field name="name">Configuration Wizards</field>
             <field name="res_model">ir.actions.todo</field>
             <field name="view_id" ref="ir_actions_todo_tree"/>
-            <field name="view_type">form</field>
             <field name="help">The configuration wizards are used to help you configure a new instance of Odoo. They are launched during the installation of new modules, but you can choose to restart some wizards manually from this menu.</field>
         </record>
         <menuitem id="menu_ir_actions_todo_form" action="act_ir_actions_todo_form" parent="base.next_id_6"/>

--- a/odoo/addons/base/views/ir_attachment_views.xml
+++ b/odoo/addons/base/views/ir_attachment_views.xml
@@ -87,7 +87,6 @@
             <field name="name">Attachments</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">ir.attachment</field>
-            <field name="view_type">form</field>
             <field name="view_id" eval="False"/>
             <field name="search_view_id" ref="view_attachment_search"/>
             <field name="help" type="html">

--- a/odoo/addons/base/views/ir_cron_views.xml
+++ b/odoo/addons/base/views/ir_cron_views.xml
@@ -79,7 +79,6 @@
         <record id="ir_cron_act" model="ir.actions.act_window">
             <field name="name">Scheduled Actions</field>
             <field name="res_model">ir.cron</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form,calendar</field>
             <field name="context">{'search_default_all': 1}</field>
             <field name="view_id" ref="ir_cron_view_tree"/>

--- a/odoo/addons/base/views/ir_default_views.xml
+++ b/odoo/addons/base/views/ir_default_views.xml
@@ -55,7 +55,6 @@
         <field name="name">User-defined Defaults</field>
         <field name="type">ir.actions.act_window</field>
         <field name="res_model">ir.default</field>
-        <field name="view_type">form</field>
         <field name="view_mode">tree,form</field>
         <field name="search_view_id" ref="ir_default_search_view"/>
     </record>

--- a/odoo/addons/base/views/ir_logging_views.xml
+++ b/odoo/addons/base/views/ir_logging_views.xml
@@ -57,7 +57,6 @@
         <record model="ir.actions.act_window" id="ir_logging_all_act">
             <field name="name">Logging</field>
             <field name="res_model">ir.logging</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
             <field name="search_view_id" ref="ir_logging_search_view" />
         </record>

--- a/odoo/addons/base/views/ir_mail_server_views.xml
+++ b/odoo/addons/base/views/ir_mail_server_views.xml
@@ -51,7 +51,6 @@
         <record model="ir.actions.act_window" id="action_ir_mail_server_list">
             <field name="name">Outgoing Mail Servers</field>
             <field name="res_model">ir.mail_server</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
             <field name="view_id" ref="ir_mail_server_list" />
             <field name="search_view_id" ref="view_ir_mail_server_search"/>

--- a/odoo/addons/base/views/ir_model_views.xml
+++ b/odoo/addons/base/views/ir_model_views.xml
@@ -224,7 +224,6 @@
         <record id="action_model_model" model="ir.actions.act_window">
             <field name="name">Models</field>
             <field name="res_model">ir.model</field>
-            <field name="view_type">form</field>
             <field name="context">{}</field>
             <field name="view_id" ref="view_model_tree"/>
         </record>
@@ -383,7 +382,6 @@
         <record id="action_model_fields" model="ir.actions.act_window">
             <field name="name">Fields</field>
             <field name="res_model">ir.model.fields</field>
-            <field name="view_type">form</field>
             <field name="context">{}</field>
             <field name="view_id" ref="view_model_fields_tree"/>
         </record>
@@ -449,7 +447,6 @@
         <record id="action_model_data" model="ir.actions.act_window">
             <field name="name">External Identifiers</field>
             <field name="res_model">ir.model.data</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="view_model_data_list"/>
         </record>
         <menuitem id="next_id_5" name="Sequences &amp; Identifiers" parent="base.menu_custom" sequence="21" groups="base.group_no_one"/>
@@ -489,7 +486,6 @@
         <record id="action_model_constraint" model="ir.actions.act_window">
             <field name="name">Model Constraints</field>
             <field name="res_model">ir.model.constraint</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="view_model_constraint_list"/>
         </record>
         <menuitem action="action_model_constraint" id="ir_model_constraint_menu" parent="base.next_id_9"
@@ -526,7 +522,6 @@
         <record id="action_model_relation" model="ir.actions.act_window">
             <field name="name">ManyToMany Relations</field>
             <field name="res_model">ir.model.relation</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="view_model_relation_list"/>
         </record>
         <menuitem action="action_model_relation" id="ir_model_relation_menu" parent="base.next_id_9"
@@ -615,7 +610,6 @@
         <record id="ir_access_act" model="ir.actions.act_window">
             <field name="name">Access Rights</field>
             <field name="res_model">ir.model.access</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="ir_access_view_tree_edition"/>
             <field name="search_view_id" ref="ir_access_view_search"/>
         </record>

--- a/odoo/addons/base/views/ir_module_views.xml
+++ b/odoo/addons/base/views/ir_module_views.xml
@@ -203,7 +203,6 @@
         <record id="open_module_tree" model="ir.actions.act_window">
             <field name="name">Apps</field>
             <field name="res_model">ir.module.module</field>
-            <field name="view_type">form</field>
             <field name="view_mode">kanban,tree,form</field>
             <field name="context">{'search_default_app':1}</field>
             <field name="search_view_id" ref="view_module_filter"/>

--- a/odoo/addons/base/views/ir_property_views.xml
+++ b/odoo/addons/base/views/ir_property_views.xml
@@ -59,7 +59,6 @@
             <field name="name">Company Properties</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">ir.property</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="ir_property_view_tree"/>
         </record>
         <menuitem id="menu_ir_property_form_all" parent="menu_ir_property" action="ir_property_form"/>

--- a/odoo/addons/base/views/ir_rule_views.xml
+++ b/odoo/addons/base/views/ir_rule_views.xml
@@ -89,7 +89,6 @@
         <record id="action_rule" model="ir.actions.act_window">
             <field name="name">Record Rules</field>
             <field name="res_model">ir.rule</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="view_rule_tree"/>
             <field name="search_view_id" ref="view_rule_search"/>
         </record>

--- a/odoo/addons/base/views/ir_sequence_views.xml
+++ b/odoo/addons/base/views/ir_sequence_views.xml
@@ -98,7 +98,6 @@
             <field name="name">Sequences</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">ir.sequence</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="sequence_view_tree"/>
             <field name="context">{'active_test': False}</field>
         </record>

--- a/odoo/addons/base/views/ir_translation_views.xml
+++ b/odoo/addons/base/views/ir_translation_views.xml
@@ -81,7 +81,6 @@
         <record id="action_translation" model="ir.actions.act_window">
             <field name="name">Translated Terms</field>
             <field name="res_model">ir.translation</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="view_translation_tree"/>
         </record>
 

--- a/odoo/addons/base/views/ir_ui_menu_views.xml
+++ b/odoo/addons/base/views/ir_ui_menu_views.xml
@@ -72,7 +72,6 @@
         <record id="grant_menu_access" model="ir.actions.act_window">
             <field name="name">Menu Items</field>
             <field name="res_model">ir.ui.menu</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="edit_menu"/>
             <field name="context">{'ir.ui.menu.full_list':True}</field>
             <field name="search_view_id" ref="edit_menu_access_search"/>

--- a/odoo/addons/base/views/ir_ui_view_views.xml
+++ b/odoo/addons/base/views/ir_ui_view_views.xml
@@ -123,7 +123,6 @@
             name="Reset View Architecture"
             binding_model="ir.ui.view" binding_views="form"
             res_model="reset.view.arch.wizard"
-            view_type="form"
             view_mode="form"
             target="new"
             groups="base.group_erp_manager"/>

--- a/odoo/addons/base/views/onboarding_views.xml
+++ b/odoo/addons/base/views/onboarding_views.xml
@@ -163,7 +163,6 @@
         <field name="type">ir.actions.act_window</field>
         <field name="res_model">res.company</field>
         <field name="view_mode">form</field>
-        <field name="view_type">form</field>
         <field name="view_id" ref="base_onboarding_company_form" />
         <field name="target">new</field>
     </record>

--- a/odoo/addons/base/views/report_paperformat_views.xml
+++ b/odoo/addons/base/views/report_paperformat_views.xml
@@ -39,13 +39,11 @@
         <record id="paper_format_action" model="ir.actions.act_window">
             <field name="name">Paper Format General Configuration</field>
             <field name="res_model">report.paperformat</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
         </record>
         <record id='reports_action' model='ir.actions.act_window'>
             <field name="name">Reports</field>
             <field name="res_model">ir.actions.report</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
         </record>
         <menuitem

--- a/odoo/addons/base/views/res_bank_views.xml
+++ b/odoo/addons/base/views/res_bank_views.xml
@@ -50,7 +50,6 @@
         <record id="action_res_bank_form" model="ir.actions.act_window">
             <field name="name">Banks</field>
             <field name="res_model">res.bank</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
             <field name="help" type="html">
               <p class="o_view_nocontent_smiling_face">
@@ -115,7 +114,6 @@
         <record id="action_res_partner_bank_account_form" model="ir.actions.act_window">
             <field name="name">Bank Accounts</field>
             <field name="res_model">res.partner.bank</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
             <field name="help" type="html">
               <p class="o_view_nocontent_smiling_face">

--- a/odoo/addons/base/views/res_company_views.xml
+++ b/odoo/addons/base/views/res_company_views.xml
@@ -98,7 +98,6 @@
             <field name="name">Companies</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.company</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
             <field name="domain">[('id','=',active_id)]</field>
         </record>
@@ -106,7 +105,6 @@
             <field name="name">Companies</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.company</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,kanban,form</field>
             <field name="help" type="html">
               <p class="o_view_nocontent_smiling_face">

--- a/odoo/addons/base/views/res_country_views.xml
+++ b/odoo/addons/base/views/res_country_views.xml
@@ -58,7 +58,6 @@
             <field name="name">Countries</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.country</field>
-            <field name="view_type">form</field>
             <field name="help">Display and manage the list of all countries that can be assigned to your partner records. You can create or delete countries to make sure the ones you are working on will be maintained.</field>
         </record>
 
@@ -94,7 +93,6 @@
             <field name="name">Country Group</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.country.group</field>
-            <field name="view_type">form</field>
             <field name="help">Display and manage the list of all countries group. You can create or delete country group to make sure the ones you are working on will be maintained.</field>
         </record>
 
@@ -145,7 +143,6 @@
             <field name="name">Fed. States</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.country.state</field>
-            <field name="view_type">form</field>
             <field name="view_id" ref="view_country_state_tree"/>
             <field name="help">You can manage the different federal states you are working on from here. Each state is attached to one country.</field>
         </record>

--- a/odoo/addons/base/views/res_currency_views.xml
+++ b/odoo/addons/base/views/res_currency_views.xml
@@ -156,7 +156,6 @@
         <record id="action_currency_form" model="ir.actions.act_window">
             <field name="name">Currencies</field>
             <field name="res_model">res.currency</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,kanban,form</field>
             <field name="search_view_id" ref="view_currency_search"/>
             <field name="context">{'active_test': False, 'search_default_active': 1}</field>
@@ -164,7 +163,6 @@
         <record id="action_currency_all_form" model="ir.actions.act_window">
             <field name="name">Currencies</field>
             <field name="res_model">res.currency</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,kanban,form</field>
             <field name="search_view_id" ref="view_currency_search"/>
             <field name="context">{'active_test': False}</field>

--- a/odoo/addons/base/views/res_lang_views.xml
+++ b/odoo/addons/base/views/res_lang_views.xml
@@ -99,7 +99,6 @@
         <record id="res_lang_act_window" model="ir.actions.act_window">
             <field name="name">Languages</field>
             <field name="res_model">res.lang</field>
-            <field name="view_type">form</field>
             <field name="context">{'active_test': False}</field>
             <field name="search_view_id" ref="res_lang_search"/>
         </record>

--- a/odoo/addons/base/views/res_partner_views.xml
+++ b/odoo/addons/base/views/res_partner_views.xml
@@ -32,7 +32,6 @@
             <field name="name">Contact Titles</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.partner.title</field>
-            <field name="view_type">form</field>
             <field name="domain">[]</field>
             <field name="context">{}</field>
             <field name="help">Manage the contact titles you want to have available in your system and the way you want to print them in letters and other documents. Some example: Mr., Mrs. </field>
@@ -548,7 +547,6 @@
             <field name="name">Customers</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.partner</field>
-            <field name="view_type">form</field>
             <field name="view_mode">kanban,tree,form</field>
             <field name="context">{"search_default_customer":1}</field>
             <field name="search_view_id" ref="view_res_partner_filter"/>
@@ -583,7 +581,6 @@
             <field name="name">Customers</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.partner</field>
-            <field name="view_type">form</field>
             <field name="view_mode">kanban,tree,form</field>
             <field name="domain">[]</field>
             <field name="context">{'default_customer':1, 'search_default_customer':1, 'default_is_company': True}</field>
@@ -619,7 +616,6 @@
             <field name="name">Vendors</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.partner</field>
-            <field name="view_type">form</field>
             <field name="domain">[]</field>
             <field name="view_mode">kanban,tree,form</field>
             <field name="context">{'search_default_supplier': 1,'default_customer': 0,'default_supplier': 1, 'default_is_company': True}</field>
@@ -656,7 +652,6 @@
             <field name="name">Employees</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.partner</field>
-            <field name="view_type">form</field>
             <field name="domain">[('employee','=',1)]</field>
             <field name="context">{'default_customer':0, 'default_supplier':0 , 'default_employee':1}</field>
             <field name="filter" eval="True"/>
@@ -666,7 +661,6 @@
             <field name="name">Other Partners</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.partner</field>
-            <field name="view_type">form</field>
             <field name="domain">[('supplier','!=',1),('customer','!=',1)]</field>
             <field name="filter" eval="True"/>
         </record>
@@ -701,7 +695,6 @@
             <field name="name">Contact Tags</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.partner.category</field>
-            <field name="view_type">form</field>
             <field name="help" type="html">
               <p class="o_view_nocontent_smiling_face">
                 Create a new contact tag
@@ -743,7 +736,6 @@
             <field name="name">Industry</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.partner.industry</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
             <field name="help" type="html">
               <p class="o_view_nocontent_smiling_face">

--- a/odoo/addons/base/views/res_users_views.xml
+++ b/odoo/addons/base/views/res_users_views.xml
@@ -31,7 +31,7 @@
             name="Change Password"
             binding_model="res.users"
             res_model="change.password.wizard"
-            view_type="form" view_mode="form" target="new"
+            view_mode="form" target="new"
             groups="base.group_erp_manager"/>
 
         <!-- res.groups -->
@@ -108,7 +108,6 @@
             <field name="name">Groups</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.groups</field>
-            <field name="view_type">form</field>
             <field name="context">{'search_default_no_share': 1}</field>
             <field name="help">A group is a set of functional areas that will be assigned to the user in order to give them access and rights to specific applications and tasks in the system. You can create custom groups or edit the ones existing by default in order to customize the view of the menu that users will be able to see. Whether they can have a read, write, create and delete access right can be managed from here.</field>
         </record>
@@ -308,7 +307,6 @@
             <field name="name">Users</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.users</field>
-            <field name="view_type">form</field>
             <field name="view_mode">tree,kanban,form</field>
             <field name="view_id" ref="view_users_tree"/>
             <field name="search_view_id" ref="view_users_search"/>
@@ -368,7 +366,6 @@
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">res.users</field>
             <field name="target">new</field>
-            <field name="view_type">form</field>
             <field name="view_mode">form</field>
         </record>
         <record id="action_res_users_my_view2" model="ir.actions.act_window.view">

--- a/odoo/addons/base/wizard/base_export_language.py
+++ b/odoo/addons/base/wizard/base_export_language.py
@@ -53,7 +53,6 @@ class BaseLanguageExport(models.TransientModel):
             'type': 'ir.actions.act_window',
             'res_model': 'base.language.export',
             'view_mode': 'form',
-            'view_type': 'form',
             'res_id': this.id,
             'views': [(False, 'form')],
             'target': 'new',

--- a/odoo/addons/base/wizard/base_export_language_views.xml
+++ b/odoo/addons/base/wizard/base_export_language_views.xml
@@ -45,7 +45,6 @@
             <field name="name">Export Translation</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">base.language.export</field>
-            <field name="view_type">form</field>
             <field name="view_mode">form</field>
             <field name="target">new</field>
         </record>

--- a/odoo/addons/base/wizard/base_import_language_views.xml
+++ b/odoo/addons/base/wizard/base_import_language_views.xml
@@ -26,7 +26,6 @@
             <field name="name">Import Translation</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">base.language.import</field>
-            <field name="view_type">form</field>
             <field name="view_mode">form</field>
             <field name="target">new</field>
         </record>

--- a/odoo/addons/base/wizard/base_language_install.py
+++ b/odoo/addons/base/wizard/base_language_install.py
@@ -37,7 +37,6 @@ class BaseLanguageInstall(models.TransientModel):
         self.state = 'done'
         return {
             'name': _('Language Pack'),
-            'view_type': 'form',
             'view_mode': 'form',
             'view_id': False,
             'res_model': 'base.language.install',

--- a/odoo/addons/base/wizard/base_language_install_views.xml
+++ b/odoo/addons/base/wizard/base_language_install_views.xml
@@ -31,7 +31,6 @@ You must change the preferences of the user to view the changes.</span>
             <field name="name">Load a Translation</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">base.language.install</field>
-            <field name="view_type">form</field>
             <field name="view_mode">form</field>
             <field name="target">new</field>
         </record>

--- a/odoo/addons/base/wizard/base_module_update.py
+++ b/odoo/addons/base/wizard/base_module_update.py
@@ -23,7 +23,6 @@ class BaseModuleUpdate(models.TransientModel):
         res = {
             'domain': str([]),
             'name': 'Modules',
-            'view_type': 'form',
             'view_mode': 'tree,form',
             'res_model': 'ir.module.module',
             'view_id': False,

--- a/odoo/addons/base/wizard/base_module_update_views.xml
+++ b/odoo/addons/base/wizard/base_module_update_views.xml
@@ -34,7 +34,6 @@
             <field name="name">Module Update</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">base.module.update</field>
-            <field name="view_type">form</field>
             <field name="view_mode">form</field>
             <field name="target">new</field>
         </record>

--- a/odoo/addons/base/wizard/base_module_upgrade_views.xml
+++ b/odoo/addons/base/wizard/base_module_upgrade_views.xml
@@ -24,7 +24,6 @@
             <field name="name">Apply Schedule Upgrade</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">base.module.upgrade</field>
-            <field name="view_type">form</field>
             <field name="view_mode">form</field>
             <field name="target">new</field>
         </record>
@@ -57,7 +56,6 @@
             <field name="name">Module Upgrade Install</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">base.module.upgrade</field>
-            <field name="view_type">form</field>
             <field name="view_mode">form</field>
             <field name="view_id" ref="view_base_module_upgrade_install"/>
             <field name="target">new</field>

--- a/odoo/addons/base/wizard/base_partner_merge_views.xml
+++ b/odoo/addons/base/wizard/base_partner_merge_views.xml
@@ -3,7 +3,6 @@
         <record id="action_partner_deduplicate" model="ir.actions.act_window">
             <field name="name">Deduplicate Contacts</field>
             <field name="res_model">base.partner.merge.automatic.wizard</field>
-            <field name="view_type">form</field>
             <field name="view_mode">form</field>
             <field name="target">new</field>
             <field name="context">{'active_test': False}</field>

--- a/odoo/addons/base/wizard/base_update_translations_views.xml
+++ b/odoo/addons/base/wizard/base_update_translations_views.xml
@@ -20,7 +20,6 @@
             <field name="name">Generate Missing Terms</field>
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">base.update.translations</field>
-            <field name="view_type">form</field>
             <field name="view_mode">form</field>
             <field name="target">new</field>
         </record>
